ts/features/walletV3/payment/screens/WalletPaymentConfirmScreen.tsx

```diff
@@ -52,51 +77,87 @@ const WalletPaymentConfirmScreen = () => {
     }
   }, [authorizationUrlPot, navigation]);
 
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const taxFee = pipe(
+    selectedPspOption,
+    O.chain(psp => O.fromNullable(psp.taxPayerFee)),
+    O.getOrElse(() => 0)
+  );
+
+  const paymentAmount = pipe(
+    pot.toUndefined(paymentDetailsPot),
+    O.fromNullable,
+    O.chain(paymentDetails => O.fromNullable(paymentDetails.amount)),
+    O.getOrElse(() => 0)
+  );
+
+  const paymentMethodDetails = pipe(
+    selectedMethodOption,
+    O.chain(method => O.fromNullable(method.details)),
+    O.toUndefined
+  );
+
+  const selectedMethod = O.toUndefined(selectedMethodOption);
+
+  const selectedPsp = O.toUndefined(selectedPspOption);
+
   const handleStartPaymentAuthorization = () => {
-    dispatch(
-      walletPaymentAuthorization.request({
-        paymentAmount: 1000 as AmountEuroCents,
-        paymentFees: 1000 as AmountEuroCents,
-        pspId: "A",
-        transactionId: "A",
-        walletId: "A"
-      })
-    );
+    if (selectedMethod && selectedPsp && pot.isSome(transactionPot)) {
+      dispatch(
+        walletPaymentAuthorization.request({
+          paymentAmount: paymentAmount as AmountEuroCents,
+          paymentFees: taxFee as AmountEuroCents,
+          pspId: selectedPsp.idBundle ?? "",
+          transactionId: transactionPot.value.transactionId,
+          walletId: selectedMethod.walletId
+        })
+      );
+    }
   };
 
-  return (
-    <BaseScreenComponent goBack={true}>
-      <GradientScrollView
-        primaryActionProps={{
-          label: "Paga xx,xx €",
-          accessibilityLabel: "Paga xx,xx €",
-          onPress: handleStartPaymentAuthorization,
-          disabled: isLoading,
-          loading: isLoading
-        }}
-      >
-        <DebugPrettyPrint
-          title="selectedMethodOption"
-          data={selectedMethodOption}
-          startCollapsed={true}
-        />
-        <VSpacer size={8} />
-        <DebugPrettyPrint
-          title="selectedPspOption"
-          data={selectedPspOption}
-          startCollapsed={true}
-        />
-        <VSpacer size={8} />
-        <DebugPrettyPrint title="transactionPot" data={transactionPot} />
-        <VSpacer size={8} />
-        <DebugPrettyPrint
-          title="authorizationUrlPot"
-          data={authorizationUrlPot}
+  if (isError) {
+    // TODO: Failure handling (https://pagopa.atlassian.net/browse/IOBP-471)
+    return <></>;
+  }
+
+  const LoadingContent = () => (
+    <SafeAreaView style={styles.loadingContainer}>
+      <LoadingSpinner size={48} />
+      <VSpacer size={24} />
+      <H3 style={{ textAlign: "center" }}>
+        {I18n.t("payment.confirm.loading.title")}
+      </H3>
+    </SafeAreaView>
+  );
+
+  return pipe(
+    sequenceS(O.Monad)({
+      paymentMethodDetails: pipe(paymentMethodDetails, O.fromNullable),
```

Comment:
a couple of times in this `sequenceS` you are passing data that was translated from option to nullable and then again to option; why not directly do the interpolation there? e.g.
```ts
sequenceS(O.Monad)({
      paymentMethodDetails: pipe(
        selectedMethodOption,
        O.chainNullableK(method => method.details)
      ),
   ```
   
   done that, I notice that you sometimes need the "nullable" iteration of said data in the handler, so I say:
   what about doing something like:
   main component checks for data existance with `sequenceS`.
   - if it is null, return loading/error component
   - if it is some, pass sequence data to a `successContent` component, so that you avoid handling too much "uncertain" data
   
   what do you think? IMHO it makes the code clearer and more "sectioned", but it very well may be personal preference up to a certain point.

Reply:
Addressed into 3843a46f0b541a3809c37104b71a24f3a568a10f